lib/OpenQA/BuildResults.pm

```diff
@@ -123,29 +123,34 @@ sub compute_build_results {
     }
 
     # split the statement - give in to perltidy
-    my $search_opts = {
-        select => ['VERSION', 'BUILD', {min => 't_created', -as => 'first_hit'}],
-        as     => [qw(VERSION BUILD first_hit)],
-        # order-by is required here despite it is not relevant when iterating results
-        # because the ordering must be applied before the row limit is applied
-        order_by => {-desc => 'first_hit'},
-        group_by => [qw(VERSION BUILD)]};
-    $search_opts->{rows} = $limit if defined($limit);
-    my $search_filter = {group_id => {in => $group_ids}};
+    my %search_opts = (
+        select   => [qw(VERSION BUILD)],
+        group_by => [qw(VERSION BUILD)],
+        rows     => (defined($limit) && $limit > 400) ? $limit : 400
```

Comment:
ah, I was reading your comment during a call. You can reverse the order to make sure you see latest id first in your 400 rows.